packages/kbn-esql-utils/src/utils/append_to_query.ts

```diff
@@ -5,9 +5,73 @@
  * in compliance with, at your election, the Elastic License 2.0 or the Server
  * Side Public License, v 1.
  */
+import { getAstAndSyntaxErrors } from '@kbn/esql-ast';
 
 // Append in a new line the appended text to take care of the case where the user adds a comment at the end of the query
 // in these cases a base query such as "from index // comment" will result in errors or wrong data if we don't append in a new line
 export function appendToESQLQuery(baseESQLQuery: string, appendedText: string): string {
   return `${baseESQLQuery}\n${appendedText}`;
 }
+
+export function appendWhereClauseToESQLQuery(
+  baseESQLQuery: string,
+  field: string,
+  value: unknown,
+  operation: '+' | '-',
+  fieldType?: string
+): string {
+  let operator = operation === '+' ? '==' : '!=';
+  let filterValue = typeof value === 'string' ? `"${value.replace(/"/g, '\\"')}"` : value;
+  // Adding the backticks here are they are needed for special char fields
+  let fieldName = `\`${field}\``;
+
+  // casting to string
+  // there are some field types such as the ip that need
+  // to cast in string first otherwise ES will fail
+  if (fieldType !== 'string' && fieldType !== 'number' && fieldType !== 'boolean') {
+    fieldName = `${fieldName}::string`;
+  }
+
+  // checking that the value is not null
+  // this is the existence filter
+  if (field === '_exists_') {
+    fieldName = `\`${String(value)}\``;
+    operator = ' is not null';
+    filterValue = '';
+  }
+
+  const { ast } = getAstAndSyntaxErrors(baseESQLQuery);
+
+  const lastCommandIsWhere = ast[ast.length - 1].name === 'where';
+  // if where command already exists in the end of the query:
+  // - we need to append with and if the filter doesnt't exist
+  // - we need to change the filter operator if the filter exists with different operator
+  // - we do nothing if the filter exists with the same operator
```

Comment:
It allowed to add `agent` twice. Is it expected?

<img width="1201" alt="Screenshot 2024-04-30 at 14 43 16" src="https://github.com/elastic/kibana/assets/1415710/60f524a6-c8ad-4fa2-986e-c4f85e41fbc5">

Reply:
Yes I didn't want to complicate it more. I dont expect many users to click the existence of the same field multiple times but I am open to discussion if you feel is necessary

Reply:
I actually did that as it was quite straightforward https://github.com/elastic/kibana/pull/181399/commits/193e6708d46b590b23b42073188ab1bbddca1e49